servant/src/Servant/Utils/Links.hs

```diff
@@ -105,11 +105,7 @@ import Data.List
 import Data.Proxy ( Proxy(..) )
 import qualified Data.Text as Text
 import qualified Data.ByteString.Char8 as BSC
-#if !MIN_VERSION_base(4,8,0)
-import Data.Monoid ( Monoid(..), (<>) )
-#else
-import Data.Monoid ( (<>) )
-#endif
+import Data.Monoid.Compat ( Monoid(..), (<>) )
```

Comment:
IIUC `Monoid` is being exported by newer `Prelude`s (and hence also by `Prelude.Compat`). `(<>)` is not being exported by newer `Prelude`s (and hence also not by `Prelude.Compat`). So what we want here, I guess, is:

``` haskell
import Prelude ()
import Prelude.Compat
import Data.Monoid ((<>))
```
The point is: less CPP.